File: src/main/resources/output/IFC4X3_java/Types/IfcSolidOrShell.java
// SELECT TYPE IfcSolidOrShell
public class IfcSolidOrShell {
    public Object value;

    public IfcSolidOrShell(Object value) {
        if (!(value instanceof IfcClosedShell || value instanceof IfcSolidModel)) {
            throw new IllegalArgumentException("Value must be one of: IfcClosedShell, IfcSolidModel");
        }
        this.value = value;
    }

    public <T> T getValueAs(Class<T> clazz) {
        return clazz.cast(value);
    }
}

File: src/main/resources/output/IFC4X3_csharp/Types/IfcSolidOrShell.cs
// SELECT TYPE IfcSolidOrShell
public class IfcSolidOrShell {
    public object Value { get; }

    public IfcSolidOrShell(object value) {
        if (!(value is IfcClosedShell || value is IfcSolidModel))
        {
            throw new ArgumentException("Value must be one of: IfcClosedShell, IfcSolidModel");
        }
        Value = value;
    }

    public T GetValueAs<T>() {
        return (T)Value;
    }
}
